mountetna.github.io/vulcan.md

```diff
@@ -0,0 +1,117 @@
+       showSum:
+         run: ui-outputs/raw.cwl
+         in:
+           a: pickANum/num
+         out: []
+
+The first step in the example makes the two primary inputs `someInt` and
+`anotherInt` available to a python script as inputs `a` and `b`. Here is the
+corresponding Python script `add.py`:
+
+    from archimedes.functions.dataflow import output_path, input_path
+
+    a = int(open(input_path('a'), 'r').read())
+    b = int(open(input_path('b'), 'r').read())
+
+    with open(output_path('sum'), 'w') as output_file:
+        output_file.write(str(a + b))
+
+When the step writes output to the appropriate file location, it becomes
+available to subsequent (dependent) steps in the workflow. It also becomes part
+of Vulcan's cache.
+
+The second step is a UI query step, which uses output from previous steps to
+request input from the user using one of Vulcan's query widgets (e.g., a text
+box with a label returning an integer), allowing the user to make a response
+and write it into the cache for subsequent steps.  There is no associated
+Python code; the Vulcan orchestrator will pass the appropriate instruction to
+the UI for `ui-queries`.
+
+The last step in the example workflow generates output instructions for the
+Vulcan UI, which can be any of a number of output artifacts: an image, a file
+download, a table, an interactive plot, etc.
+
+### Vulcan Cache
+
+The cache is an ephemeral data store that holds all of the output produced by
+running workflows. Each step in the workflow, when it completes, stores data in
+a cache location.  On subsequent runs, if there is already a cached output for
+the step, it will be used instead of rerunning the step.  By making use of
+cached intermediates, Vulcan workflows can minimize the amount of calculation
+required by iteration.
+
+If the inputs or the script for a step change, the location of the step cache
+will also change, and Vulcan will look for output in the new location. Note
+that the "new" location may also be a previously cached state (e.g., if you
+toggle back and forth between two input values). In addition each workflow
+session contains a unique key, which if reset will change the cache locations.
+
+### Running a Workflow
+
+Workflows can be run in the Vulcan UI. To run a workflow, navigate to your
+project page from the Vulcan root page. Select one of the available workflows
+and the Vulcan Workflow UI will appear with a new workflow session.
+
+The interface is divided into three columns. On the left, the user may
+configure primary inputs, or, once the workflow is running, UI query inputs
+created by the workflow. Outputs, such as plots or download links, appear in
+the second column. The final column shows the steps in the workflow in the
+order they will run (approximately, since the workflow may branch).
+
+The user may change inputs as they see fit; changes will not be persisted to
+the session until the user commits their edits to each required input. Once the
```

Comment:
Would it be helpful to explain the `commit` button here? I think we had talked about changing how that interaction would happen (commit vs run), not sure if we reached a concrete conclusion...